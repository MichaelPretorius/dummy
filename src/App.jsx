import React from 'react';

import { GlobalStyle } from './theme/globalStyles';

function App() {
  return (
    <>
      <GlobalStyle />
      <div className="App">App</div>
    </>
  );
}

export default App;
