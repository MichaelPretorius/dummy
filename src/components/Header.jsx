import React from 'react';
import { Link, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';

import { selectIsAuthenticated } from '../redux/auth/authSelectors';
import { NormalButton } from './CustomButton';
import { logoutUser } from '../redux/auth/authActions';

const Header = ({ isAuthenticated, logoutUser, history }) => {
  return (
    <>
      <Link to="/">Home</Link>
      <Link to="/private">Private</Link>
      {isAuthenticated ? (
        <NormalButton onClick={() => logoutUser(() => history.push('/'))}>
          Logout
        </NormalButton>
      ) : (
        <>
          <Link to="/signup">Signup</Link>
          <Link to="/login">Login</Link>
        </>
      )}
    </>
  );
};

const mapStateToProps = state => ({
  isAuthenticated: selectIsAuthenticated(state),
});

export default withRouter(
  connect(
    mapStateToProps,
    { logoutUser }
  )(Header)
);
