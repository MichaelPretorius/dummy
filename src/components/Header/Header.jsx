import React, { useState } from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';

import { selectIsAuthenticated } from '../../redux/auth/authSelectors';
import { logoutUser } from '../../redux/auth/authActions';
import burger from '../../assets/navburger.png';
import {
  Button,
  NavContainer,
  Wrapper,
  Burger,
  NavBrand,
  Navbar,
  Navs,
  Nav,
} from '../../styles';

export const Header = ({ isAuthenticated, logoutUser, history }) => {
  const [showNav, setShowNav] = useState(false);

  const toggleNav = () => {
    setShowNav(!showNav);
  };

  return (
    <NavContainer row secondary>
      <Wrapper row>
        <Burger onClick={toggleNav} src={burger} />
        <NavBrand to="/">Home</NavBrand>
      </Wrapper>
      <Navbar showNav={showNav} width="200px">
        <Navs>
          <Nav to="/styles">Styles</Nav>
          <Nav to="/private">Private</Nav>
          {isAuthenticated ? (
            <Button onClick={() => logoutUser(() => history.push('/'))}>
              Logout
            </Button>
          ) : (
            <>
              <Nav to="/signup">Signup</Nav>
              <Nav to="/login">Login</Nav>
            </>
          )}
        </Navs>
      </Navbar>
    </NavContainer>
  );
};

const mapStateToProps = state => ({
  isAuthenticated: selectIsAuthenticated(state),
});

export default withRouter(
  connect(
    mapStateToProps,
    { logoutUser }
  )(Header)
);
