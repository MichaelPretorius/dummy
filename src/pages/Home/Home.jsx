import React from 'react';

import { Header } from '../../styles';

export const Home = () => {
  return (
    <Header center secondary>
      Home
    </Header>
  );
};

export default Home;
