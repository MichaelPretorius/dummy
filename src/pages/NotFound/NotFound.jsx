import React from 'react';

import { Header } from '../../styles';

export const NotFound = () => {
  return (
    <Header center secondary>
      NotFound
    </Header>
  );
};

export default NotFound;
