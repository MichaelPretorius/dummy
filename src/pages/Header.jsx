import React from 'react';
import { Link, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';

import { selectCurrentUser } from '../redux/auth/authSelectors';
import { NormalButton } from '../components/CustomButton';
import { logoutUser } from '../redux/auth/authActions';

const Header = ({ user, logoutUser, history }) => {
  return (
    <>
      <Link to="/">Home</Link>
      <Link to="/private">Private</Link>
      {user ? (
        <NormalButton onClick={() => logoutUser(() => history.push('/'))}>
          Logout
        </NormalButton>
      ) : (
        <>
          <Link to="/signup">Signup</Link>
          <Link to="/login">Login</Link>
        </>
      )}
    </>
  );
};

const mapStateToProps = state => ({
  user: selectCurrentUser(state),
});

export default withRouter(
  connect(
    mapStateToProps,
    { logoutUser }
  )(Header)
);
