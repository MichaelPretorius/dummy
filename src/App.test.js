import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import { App } from './App';
import Header from './components/Header/Header';
import Home from './pages/Home/Home';
import NotFound from './pages/NotFound/NotFound';

const mockStore = configureStore();
const store = mockStore({
  auth: {
    isAuthenticated: true,
  },
});

describe('App component', () => {
  let wrapper;
  const mockProps = {
    initialFetch: true,
  };
  beforeEach(() => {
    wrapper = shallow(<App {...mockProps} />);
  });

  test('should render App component', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  test('should render Header with page if initialFetch is true', () => {
    expect(wrapper.exists(Header)).toEqual(true);
  });

  test('should render Loading... if initialFetch is false', () => {
    const mockProps1 = {
      initialFetch: false,
    };
    wrapper = shallow(<App {...mockProps1} />);
    expect(wrapper.contains(<div>Loading...</div>)).toEqual(true);
  });

  test('should render NotFound Page for invalid path', () => {
    wrapper = mount(
      <MemoryRouter initialEntries={['/random']}>
        <Provider store={store}>
          <App {...mockProps} />
        </Provider>
      </MemoryRouter>
    );
    expect(wrapper.find(Home)).toHaveLength(0);
    expect(wrapper.find(NotFound)).toHaveLength(1);
  });

  test('should render component(Home) for valid path(/)', () => {
    wrapper = mount(
      <MemoryRouter initialEntries={['/']}>
        <Provider store={store}>
          <App {...mockProps} />
        </Provider>
      </MemoryRouter>
    );
    expect(wrapper.find(Home)).toHaveLength(1);
    expect(wrapper.find(NotFound)).toHaveLength(0);
  });
});
