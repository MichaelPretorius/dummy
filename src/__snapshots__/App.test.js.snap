// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render App component 1`] = `
"<Fragment>
  <GlobalStyleComponent />
  <withRouter(Connect(Header)) />
  <Switch>
    <Route exact={true} path=\\"/\\" component={[Function: Home]} />
    <Route exact={true} path=\\"/signup\\" component={{...}} />
    <Route exact={true} path=\\"/login\\" component={{...}} />
    <Connect(AuthRoute) exact={true} path=\\"/private\\" component={[Function: Private]} />
    <Route path=\\"*\\" component={[Function: NotFound]} />
  </Switch>
</Fragment>"
`;
